cmd/snap-confine/snap-confine.c

```diff
@@ -124,29 +117,11 @@ int main(int argc, char **argv)
 		return 0;
 	}
 
-	const char *snap_instance = getenv("SNAP_INSTANCE_NAME");
-	if (snap_instance == NULL) {
-		die("SNAP_INSTANCE_NAME is not set");
-	}
-	sc_instance_name_validate(snap_instance, NULL);
-
-	// Collect and validate the security tag and a few other things passed on
-	// command line.
-	const char *security_tag = sc_args_security_tag(args);
-	if (!verify_security_tag(security_tag, snap_instance)) {
-		die("security tag %s not allowed", security_tag);
-	}
-	const char *executable = sc_args_executable(args);
-	const char *base_snap_name = sc_args_base_snap(args) ? : "core";
-	bool classic_confinement = sc_args_is_classic_confinement(args);
-
-	sc_snap_name_validate(base_snap_name, NULL);
-
-	debug("security tag: %s", security_tag);
-	debug("executable:   %s", executable);
-	debug("confinement:  %s",
-	      classic_confinement ? "classic" : "non-classic");
-	debug("base snap:    %s", base_snap_name);
+	/* Collect all invocation parameters. This gives us authoritative
+	 * information about what needs to be invoked and how. The data comes
+	 * from either the environment or from command line arguments */
+	sc_invocation SC_CLEANUP(sc_cleanup_invocation) invocation;
+	sc_init_invocation(&invocation, args, getenv("SNAP_INSTANCE_NAME"));
```

Comment:
See above, I would move getenv() to a different line with a extra check for NULL or move it inside sc_init_invocation().

Reply:
Done now.